src/test/regress/citus_tests/test/test_other_databases.py

```diff
@@ -0,0 +1,140 @@
+def test_main_commited_outer_not_yet(cluster):
+    c = cluster.coordinator
+
+    # create a non-main database
+    c.sql("CREATE DATABASE db1")
+
+    # we will use cur1 to simulate non-main database user and
+    # cur2 to manually do the steps we would do in the main database
+    with c.cur(dbname="db1") as cur1:
+        with c.cur() as cur2:
```

Comment:
nit: you can combine these two in one with statements, which reduces the indentation of the following code:
```suggestion
    with c.cur(dbname="db1") as cur1, \
        c.cur() as cur2:
```

P.S. I learned of this only a few months ago